### PR TITLE
Final product

### PR DESCRIPTION
**Pull request by troyerjl**

### CODE REVIEW

File: dotm_search.py [hunk 1]
```diff
@@ -5,12 +5,54 @@
...
+import argparse
+import os
+import sys
 
 
+# def open_function(args):
+#     full_path = args
+#     with ZipFile(full_path) as z:
+#         with z.open('word/document.xml') as dotm_doc:
+#             print(dotm_doc.read())
```
Comment (knmarvel) [inline dotm_search.py:21]: Remove commented out code for a final version.

File: dotm_search.py [hunk 2]
```diff
@@ -5,12 +5,54 @@
...
+        full_path = os.path.join(args.dir, file)
+
+        with zipfile.ZipFile(full_path) as z:
+            with z.open("word/document.xml") as doc:
+                for line in doc:
+                    text_position = line.find(args.text)
+                    if text_position >= 0:
+                        match_count += 1 
+                        print("Match found in file {}".format(file))
+                        print("...{}...".format(line[text_position-40:text_position+40]))
```
Comment (knmarvel) [inline dotm_search.py:53]: Oh, I like this format method.

File: dotm_search.py [hunk 3]
```diff
@@ -5,12 +5,54 @@
...
+#     full_path = args
+#     with ZipFile(full_path) as z:
+#         with z.open('word/document.xml') as dotm_doc:
+#             print(dotm_doc.read())
+
+
+def create_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--dir", default=".", help="read .dotm files from a directory")
+    parser.add_argument("text")
```
Comment (Janell-Huyck) [inline dotm_search.py:27]: Do we not have to put in an action,  like "action = 'store'" if we just want something saved?

File: dotm_search.py [hunk 4]
```diff
@@ -5,12 +5,54 @@
...
 
 
+# def open_function(args):
+#     full_path = args
+#     with ZipFile(full_path) as z:
+#         with z.open('word/document.xml') as dotm_doc:
+#             print(dotm_doc.read())
+
+
+def create_parser():
```
Comment (Janell-Huyck) [inline dotm_search.py:24]: nice job putting this in its own function here.  makes things easy to read.

File: dotm_search.py [hunk 5]
```diff
@@ -5,12 +5,54 @@
...
+    parser.add_argument("--dir", default=".", help="read .dotm files from a directory")
+    parser.add_argument("text")
+    return parser
+
+
+def main(args):
+    parser = create_parser()
+    args = parser.parse_args()
+    
+    file_list = os.listdir(args.dir)
```
Comment (Janell-Huyck) [inline dotm_search.py:35]: clever. i never thought of sending args.dir in as a parameter - i've been sending individual arguments.  hmmm :thinking:

File: dotm_search.py [hunk 6]
```diff
@@ -5,12 +5,54 @@
...
+    match_count = 0
+    search_count = 0
+
+    for file in file_list:
+        if not file.endswith(".dotm"):
+            print("Disregarding file {}".format(file))
```
Comment (Janell-Huyck) [inline dotm_search.py:41]: At least in ours, this wasn't part of the instructions.  Extra prints like that can clutter up your screen - use judiciously :)

File: dotm_search.py (hunk 2 above)
Comment (Janell-Huyck) [reply dotm_search.py:53]: Agreed, good use of format here!

**Pull request closed without merging**